lib/assert.js

```diff
@@ -140,90 +140,149 @@ assert.deepStrictEqual = function deepStrictEqual(actual, expected, message) {
   }
 };
 
+function areSimilarRegExps(a, b) {
+  return a.source === b.source &&
+         a.global === b.global &&
+         a.multiline === b.multiline &&
+         a.lastIndex === b.lastIndex &&
```

Comment:
So, this is not related to the changes you are making here, but I would have a couple of suggestions:

- I guess you could probably make arguments both for including `lastIndex` here and for leaving it out, and I’m tempted to say “drop it”. I probably wouldn’t expect this.

- Maybe instead of the `global`, `multiline` and `ignoreCase` checks a single check for `.flags` would be a good idea? There are new flags like `u` that would currently be ignored.

(If you think these suggestions make sense, it’s probably best to keep that for a separate commit/separate PR.)